### PR TITLE
[cli] [fix] Fix display of OT node log information

### PR DESCRIPTION
**Pull request by EskoDijk**
OT firmware with full logs enabled wasn't working; this is now fixed. For a usable experience at the "Info" log level, detailed messages are only shown when in the context of the node.  (This may have different requirements when running from script.)

It now uses the new style of short log level strings as used by OT ([C], [I], etc.)

Idea is that in simulation/node.go in the future maybe different logging policies could be applied, depending on the need. E.g. some users may have a need to write all OT node's logs to individual files for later analysis.

Also the log level of the "Watch" function is fixed - instead of warning it now uses Info.

### CODE REVIEW

**Comment by EskoDijk** [comment]
Just added code for the proposal that if a user types "watch 1" in the CLI, then all the OT debug/info/note messages of Node 1 would be shown - regardless of the current CLI's node context.  This works well with the watch command in #348. Use case example: set watch on node 1, then do a "ping 1 2" without being in node 1 context.

File: otoutfilter/OTOutFilter.go [hunk 1]
```diff
@@ -35,13 +36,14 @@ import (
 )
 
 var (
-	logPattern = regexp.MustCompile(`\[(NONE|CRIT|WARN|NOTE|INFO|DEBG)].*\n`)
+	logPattern = regexp.MustCompile(`\d\d:\d\d:\d\d\.\d\d\d \[(-|C|W|N|I|D)].*\n`)
```
Comment (simonlingoogle) [inline otoutfilter/OTOutFilter.go:39]: Are we sure that the OT will compile with `OPENTHREAD_CONFIG_LOG_PREPEND_UPTIME=1`?

File: simulation/types.go [hunk 2]
```diff
@@ -1,7 +1,14 @@
 package simulation
 
-import "io"
+import (
+	"github.com/openthread/ot-ns/types"
```
Comment (simonlingoogle) [inline simulation/types.go:4]: Needs to use `goimports` to order the imported modules. @EskoDijk

File: simulation/node.go [hunk 3]
```diff
@@ -614,6 +614,25 @@ func (node *Node) lineReader(reader io.Reader, uartType NodeUartType) {
...
 
+// handles an incoming log message from the OT-Node. The policy may be changed in the future,
+// or made configurable.
+func (node *Node) handlerLogMsg(otLevel string, msg string) {
+	switch otLevel {
+	case "D", "I", "N":
+		// only display detailed log msgs if in Node's context or watching that Node.
+		// Otherwise, too much is displayed.
+		if node.S.cmdRunner.GetContextNodeId() == node.Id || node.S.Dispatcher().IsWatching(node.Id) {
+			simplelogger.Infof(msg)
```
Comment (simonlingoogle) [inline simulation/node.go:625]: Should use `Debugf` for "D" ?

File: otoutfilter/OTOutFilter.go [hunk 4]
```diff
@@ -96,7 +98,9 @@ func (cc *otOutFilter) readFirstLine(p []byte) int {
 				// remove the log
 				simplelogger.AssertTrue(logIdx[1] == len(firstline))
 				logStr := strings.TrimSpace(firstline)
-				cc.printLog(logStr)
+				logIdxStr := logPattern.FindStringSubmatch(firstline)
```
Comment (simonlingoogle) [inline otoutfilter/OTOutFilter.go:101]: Matching the regex twice seems unefficient. 

Can we call `FindStringSubmatch` or `FindStringSubmatchIndex` once?

File: otoutfilter/OTOutFilter.go (hunk 1 above)
Comment (EskoDijk) [reply otoutfilter/OTOutFilter.go:39]: Probably not, timestamp can be included as optional match.

File: simulation/node.go (hunk 3 above)
Comment (EskoDijk) [reply simulation/node.go:625]: Yes this makes most sense, but I had the problem that OTNS-internal logging with 'debug' level then also appears which makes it hard to inspect what the OT-node is doing.  
So ideally I want to be able to see OT-node debug level, while not seeing OTNS debug level messages.

Comment (simonlingoogle) [reply simulation/node.go:625]: I think the real issue here might be that we are using a single log level for both node logs and OTNS logs. We may even have another log level for watched nodes.

File: otoutfilter/OTOutFilter.go (hunk 1 above)
Comment (EskoDijk) [reply otoutfilter/OTOutFilter.go]: Latest commit simplifies this by not checking timestamp details. Any line detected as a "log line" is now sent to the printLog() function and filtered out as regular node output.

File: simulation/types.go (hunk 2 above)
Comment (EskoDijk) [reply simulation/types.go]: Fixed in an earlier commit. Sometimes the issue can come back to the automatic IDE actions.

File: simulation/node.go (hunk 3 above)
Comment (EskoDijk) [reply simulation/node.go:630]: For now this is resolved by using "debug" level for OT node "debug" messages. Having separate levels is indeed the most useful - saving this up for a future PR.

**Pull request closed without merging**